cloudwatch-kinesis/variables.tf

```diff
@@ -0,0 +1,67 @@
+locals {
+  ecs_cluster_name = var.ecs_cluster_name == null ? "${var.project}-${var.environment}" : var.ecs_cluster_name
+  name             = "${var.project}-${var.environment}-http-endpoint"
+}
+
+variable "environment" {
+  type = string
+}
+
+variable "project" {
+  type = string
+}
+
+# Regional allows clusters with the same name to be in multiple regions
+variable "region" {
+  type    = string
+  default = "eu-central-1"
+}
+
+variable "ecs_cluster_name" {
+  type    = string
+  default = null
+}
+
+variable "ecs_http_port" {
+  type    = number
+  default = 5073
+}
+
+variable "log_bucket_arn" {
+  type        = string
+  description = "destination s3 bucket which will store the logs"
+}
+
+variable "buffer_size_for_s3" {
+  type        = number
+  default     = 10
+  description = "Buffer incoming data to the specified size, in MBs, before delivering it to the s3 bucket."
+}
+
+variable "buffer_interval_for_s3" {
+  type        = number
+  default     = 400
```

Comment:
```suggestion
  default     = 1800 # 30 min
```

1 file every 6-7 minutes is already a lot, especially if we run a lot of tasks. Imagine we run 10 tasks, each logging 1 file every 6.5 minutes:

10 tasks * 1 file / 6.5 minutes * 1440 minutes/day * 30.25 days/month = 67.000 files per month.

Can we prefix the path for the files in the S3 bucket? With vector, we had the following prefix:

```
"${RAILS_ENV}/%Y-%m/%d/%H/%F-%H:%M-${RAILS_ENV}-"
```

resulting in:
* 1 folder per environment, 1 folder per year/month, 1 folder per day, 1 folder per hour
* files are prefixed with the timestamp and the environment

that should make it fairly easy to retrieve log files manually for a given time period. WDYT?

Reply:
We might not have a file generated for each task because the file is only generated when there is an error in http-endpoint/processing failed

![Screenshot 2023-05-04 at 12 19 55 PM](https://user-images.githubusercontent.com/15337955/236111807-030a89d4-9b3b-4f75-890f-45d151c00cbb.png)

In fact, I think we should have a shorter time to get an alert as soon as possible when the error happens. (Perhaps to send Slack message when a new file is added to the bucket?)

> Can we prefix the path for the files in the S3 bucket? 

Yes!

![Screenshot 2023-05-04 at 12 23 02 PM](https://user-images.githubusercontent.com/15337955/236112078-9f066869-0b78-487f-a185-1eae74a1b840.png)

Reply:
@swiknaba 
Should we also backup the logs in S3?